packages/react/presence/src/Presence.tsx

```diff
@@ -69,47 +68,53 @@ function usePresence(present: boolean) {
   }, [node]);
 
   React.useEffect(() => {
-    return waitForAfterNextFrame(() => {
+    let cancelWaitForAfterNextFrame = () => {};
+    const prevAnimationName = prevAnimationNameRef.current;
+    const currentAnimationName = getAnimationName(styles);
+    prevAnimationNameRef.current = currentAnimationName;
+
+    if (present) {
+      send('MOUNT');
+    } else if (styles?.display === 'none') {
+      // If the element is hidden, animations won't run so we unmount instantly
+      send('UNMOUNT');
+    } else {
       /**
-       * When `present` changes, we check changes to animation-name to determine
-       * whether an animation has started. We chose this approach (reading computed
-       * styles) over using `animationstart` event because we would need to delay
-       * unmount with a `setTimeout` that `animationstart` would clear. That approach
-       * becomes problematic if consumers use `animation-delay`.
+       * When `present` changes to `false`, we check changes to animation-name to
+       * determine whether an animation has started. We chose this approach (reading
+       * computed styles) because there is no `animationrun` event and `animationstart`
+       * fires after `animation-delay` has expired which would be too late.
        */
-      const wasPresent = prevPresentRef.current;
-      const prevAnimationName = prevAnimationNameRef.current;
-      const currentAnimationName = getAnimationName(styles);
-      const isAnimating = String(prevAnimationName) !== String(currentAnimationName);
-      prevAnimationNameRef.current = currentAnimationName;
-      prevPresentRef.current = present;
-
-      if (present) {
-        send('MOUNT');
-      } else if (wasPresent && isAnimating) {
-        // If the element is hidden, it will not run the animation
-        send(styles?.display === 'none' ? 'UNMOUNT' : 'ANIMATION_OUT');
+      const isAnimating = prevAnimationName !== currentAnimationName;
+      if (isAnimating) {
```

Comment:
Yep, it turned out that adding the `MOUNT` event to the `unmountSuspended` phase allowed me to clear a bunch of that stuff up. I always kinda wondered why we needed to check `wasPresent` because if `present` **changes** to `false` then it was implicitly `true`.